src/vs/workbench/contrib/terminal/browser/terminalService.ts

```diff
@@ -200,19 +202,67 @@ export class TerminalService implements ITerminalService {
 		this._connectionState = TerminalConnectionState.Connected;
 		// now that terminals have been restored,
 		// attach listeners to update remote when terminals are changed
-		this.attachRemoteListeners();
+		this.attachListeners(true);
 		this._onDidChangeConnectionState.fire();
 	}
 
-	private attachRemoteListeners(): void {
+	private _reconnectToLocalTerminals(): void {
+		// Reattach to all local terminals
+		const defaultLayoutInfo = JSON.stringify({ tabs: [{ terminals: [], isActive: true }] });
+		const result = this._storageService.get('localTerminalLayoutInfo', StorageScope.WORKSPACE, defaultLayoutInfo);
+		const layoutInfo = JSON.parse(result);
+
+		if (layoutInfo.tabs.length === 0) {
```

Comment:
We are checking 'if (layoutInfo) {' on line 222 but accessing its properties before this check occurs.  Is this right?

Reply:
thanks for the review, but this is not close to being merged/is a draft for that reason.

Reply:
Ooops, missed the draft status.